tests/components/tado/test_config_flow.py

```diff
@@ -260,3 +260,120 @@ async def test_form_homekit(hass: HomeAssistant) -> None:
         ),
     )
     assert result["type"] == "abort"
+
+
+async def test_import_step(hass: HomeAssistant) -> None:
+    """Test import step."""
+    mock_tado_api = _get_mock_tado_api(getMe={"homes": [{"id": 1, "name": "myhome"}]})
+
+    with patch(
+        "homeassistant.components.tado.config_flow.Tado",
+        return_value=mock_tado_api,
+    ), patch(
+        "homeassistant.components.tado.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                "username": "test-username",
+                "password": "test-password",
+                "home_id": 1,
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["data"] == {
+        "username": "test-username",
+        "password": "test-password",
+        "home_id": "1",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_step_existing_entry(hass: HomeAssistant) -> None:
+    """Test import step with existing entry."""
+    with patch(
+        "homeassistant.components.tado.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        entry = MockConfigEntry(
+            domain=DOMAIN,
+            data={
+                "username": "test-username",
+                "password": "test-password",
+                "home_id": 1,
+            },
+        )
+        entry.add_to_hass(hass)
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                "username": "test-username",
+                "password": "test-password",
+                "home_id": 1,
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+    assert len(mock_setup_entry.mock_calls) == 0
+
+
+async def test_import_step_validation_failed(hass: HomeAssistant) -> None:
+    """Test import step with validation failed."""
+    with patch(
+        "homeassistant.components.tado.config_flow.validate_input",
```

Comment:
Don't patch our own code. Patch the library. We want to test our code in the config flow.